openjpa-kernel/src/main/java/org/apache/openjpa/event/TCPRemoteCommitProvider.java

```diff
@@ -782,15 +816,17 @@ private HostAddress(String host)
             } catch (PrivilegedActionException pae) {
                 throw (UnknownHostException) pae.getException();
             }
+            GenericObjectPoolConfig<Socket> cfg = new GenericObjectPoolConfig<>();
+            cfg.setMaxTotal(_maxActive);
+            cfg.setBlockWhenExhausted(true);
+            cfg.setMaxWaitMillis(-1L);
             // -1 max wait == as long as it takes
-            _socketPool = new GenericObjectPool
-                (new SocketPoolableObjectFactory(), _maxActive,
-                    GenericObjectPool.WHEN_EXHAUSTED_BLOCK, -1);
+            _socketPool = new GenericObjectPool<>(new SocketPoolableObjectFactory(), cfg);
             _isAvailable = true;
         }
 
         private void setMaxActive(int maxActive) {
-            _socketPool.setMaxActive(maxActive);
+            _socketPool.setMaxTotal(maxActive);
```

Comment:
Why this change? Is `setMaxActive()`not available anymore?

Reply:
Yep, it is replaced with `setMaxTotal`

Reply:
I see: would then make sense to rename the enclosing method accordingly?

Reply:
I believe this is the decision OpenJPA dev's need to make ...
I tried to make changes minimal.

Reply:
Ok, I was comparing [1.6](https://commons.apache.org/proper/commons-pool/api-1.6/org/apache/commons/pool/impl/GenericObjectPool.html#setMaxActive(int))'s with [2.6.0](https://commons.apache.org/proper/commons-pool/api-2.6.0/index.html)'s javadocs and the description is effectively the same.